src/services/codefixes/convertFunctionToEs6Class.ts

```diff
@@ -0,0 +1,191 @@
+/* @internal */
+namespace ts.codefix {
+    registerCodeFix({
+        errorCodes: [Diagnostics.Convert_function_0_to_ES6_class.code],
+        getCodeActions,
+        createCodeFixDiagnosticIfApplicable
+    });
+
+    function createCodeFixDiagnosticIfApplicable(node: Node, context: CodeFixDiagnoseContext): Diagnostic | undefined {
+        if (!isSourceFileJavaScript(context.boundSourceFile)) {

```

Comment:
this is not going to scale as we add code fixes. we can not call every code fix on every node on the tree, we need a more efficient way. 
We could make codeFix register a SyntaxKind it is interested in, this will save us the dispatch on every node at least.
Alternatively, we can have the code fix return a Boolean whether to keep walking or not, this way we avoid walking expressions that we do not care about.